clients/admin-ui/src/features/integrations/ConfigureIntegrationForm.tsx

```diff
@@ -0,0 +1,209 @@
+import { Button, ButtonGroup, useToast, VStack } from "fidesui";
+import { Form, Formik } from "formik";
+import * as Yup from "yup";
+
+import {
+  CustomSelect,
+  CustomTextArea,
+  CustomTextInput,
+} from "~/features/common/form/inputs";
+import { getErrorMessage } from "~/features/common/helpers";
+import {
+  usePatchDatastoreConnectionMutation,
+  useUpdateDatastoreConnectionSecretsMutation,
+} from "~/features/datastore-connections";
+import { formatKey } from "~/features/datastore-connections/add-connection/helpers";
+import {
+  useGetAllSystemsQuery,
+  usePatchSystemConnectionConfigsMutation,
+} from "~/features/system";
+import {
+  AccessLevel,
+  ConnectionConfigurationResponse,
+  ConnectionType,
+} from "~/types/api";
+import { isErrorResult } from "~/types/errors";
+
+type FormValues = {
+  name: string;
+  keyfile_creds?: string;
+  description: string;
+  system_fides_key?: string;
+};
+
+const validationSchema = Yup.object().shape({
+  name: Yup.string().required().label("Name"),
+  keyfile_creds: Yup.string().nullable().label("Keyfile credentials"),
+  description: Yup.string().nullable().label("Description"),
+});
+
+const ConfigureIntegrationForm = ({
+  connection,
+  onCancel,
+}: {
+  connection?: ConnectionConfigurationResponse;
+  onCancel: () => void;
+}) => {
+  const [
+    updateConnectionSecretsMutationTrigger,
+    { isLoading: secretsIsLoading },
+  ] = useUpdateDatastoreConnectionSecretsMutation();
+  const [patchDatastoreConnectionsTrigger, { isLoading: patchIsLoading }] =
+    usePatchDatastoreConnectionMutation();
+  const [patchSystemConnectionsTrigger, { isLoading: systemPatchIsLoading }] =
+    usePatchSystemConnectionConfigsMutation();
+
+  const { data: allSystems } = useGetAllSystemsQuery();
+
+  const systemOptions = allSystems?.map((s) => ({
+    label: s.name ?? s.fides_key,
+    value: s.fides_key,
+  }));
+
+  const isLoading = secretsIsLoading || patchIsLoading || systemPatchIsLoading;
+
+  const initialValues: FormValues = {
```

Comment:
Note that keyfile credentials aren't included in the initial values even if the user is editing-- the API response just returns `"**********"` and not populating the form with that is consistent with behavior in the other connection forms.

Reply:
Thanks for clarifying. Can we a small comment in the code indicating that?